src/recipes/is-array.recipe.ts

```diff
@@ -1,38 +1,18 @@
-import { ArrayMinSize, ArrayMaxSize, IsArray, ValidateNested } from 'class-validator';
+import { ValidateNested } from 'class-validator';
 
 import { TypePropertyDecorator } from '../adapters';
 import { Builder } from '../builder';
-import { ArrayOptions, ArraySizeOptions, Initializer } from '../interfaces';
+import { Initializer } from '../interfaces';
+import { ArrayOptions } from '../options';
+import {
+    buildArrayPropertyDecorators,
+    buildValidatePropertyDecorator,
+} from './common.recipe';
 
 export function isObject<T>(value: T): boolean {
     return value !== null && typeof value === 'object';
 }
 
-export function buildArrayPropertyDecorators(
-    options?: boolean | ArraySizeOptions,
-): PropertyDecorator[] {
-    const decorators: PropertyDecorator[] = [];
-
-    if (!options) {
-        return decorators;
-    }
-
-    // validate data as array
-    decorators.push(IsArray());
-
-    // maybe: add a maximum size
-    if (options !== true && options.maxItems !== undefined) {
-        decorators.push(ArrayMaxSize(options.maxItems));
-    }
-
-    // maybe: add a minimum size
-    if (options !== true && options.minItems !== undefined) {
-        decorators.push(ArrayMinSize(options.minItems));
-    }
-
-    return decorators;
-}
-
 export function IsArrayRecipe<Options>(
```

Comment:
The logic for `isArray` is different from the other decorators; it'd be good to deprecate it now that we have an alternative.